### PR TITLE
Ps upgrade

### PR DESCRIPTION
**Pull request by WhiteFox08**
Removes the QueueWorkers entirely 
mob controller uses scheck()

This requires Btime.dll which handles timing. drop it in the same folder as the dme.
https://dl.dropboxusercontent.com/u/20521740/ShareX/2015/09/btime.dll

### CODE REVIEW

**Comment by Chronitonity** [comment]
Does this break anything you idiot

**Comment by WhiteFox08** [comment]
I can break my dick up your ass if that's what you're into.

As far as I can tell, no.

**Comment by lcass** [comment]
Lemme check this over

File: code/controllers/Processes/machinery.dm [hunk 1]
```diff
@@ -11,7 +11,7 @@
 		if(i > machines.len)
 			break
 		var/obj/machinery/M = machines[i]
-		if(istype(M) && !M.gcDestroyed)
+		if(istype(M))
```
Comment (ghost) [inline code/controllers/Processes/machinery.dm]: why remove the gcDestroyed check?

File: code/world.dm [hunk 2]
```diff
@@ -91,6 +91,7 @@
 
 	makepowernets()
 
+
```
Comment (ghost) [inline code/world.dm]: ???

File: code/controllers/master_controller.dm [hunk 3]
```diff
@@ -66,6 +66,9 @@ var/global/pipe_processing_killed = 0
 	//if(!emergency_shuttle)			emergency_shuttle = new /datum/shuttle_controller/emergency_shuttle()
 	if(!supply_shuttle)				supply_shuttle = new /datum/controller/supply_shuttle()
 
+
+
```
Comment (ghost) [inline code/controllers/master_controller.dm]: ???

File: code/controllers/configuration.dm [hunk 4]
```diff
@@ -42,6 +42,7 @@
 	var/allow_holidays = 0				//toggles whether holiday-specific content should be used
 	var/proxybuster = 0 //For using the proxybuster
 
+
```
Comment (ghost) [inline code/controllers/configuration.dm]: ???

File: code/controllers/Processes/nanoui.dm [hunk 5]
```diff
@@ -1,11 +1,12 @@
...
 	name = "nanoui"
-	schedule_interval = 20 // every 2 seconds
-	updateQueueInstance = new
+
 
 /datum/controller/process/nanoui/doWork()
-	updateQueueInstance.init(nanomanager.processing_uis, "process")
-	updateQueueInstance.Run()
+	for(var/p in nanomanager.processing_uis)
+		p:process()
```
Comment (ghost) [inline code/controllers/Processes/nanoui.dm:12]: scheck

File: code/controllers/Processes/mob.dm [hunk 6]
```diff
@@ -1,20 +1,16 @@
...
+		mob_list = list()
 
 /datum/controller/process/mob/doWork()
-	if(updateQueueInstance)
-		updateQueueInstance.init(mob_list, "Life")
-		updateQueueInstance.Run()
+	if (mob_list)
+		for(var/mob/M in mob_list)
+			M:Life()
+			scheck()
```
Comment (ghost) [inline code/controllers/Processes/mob.dm]: This doesn't look very safe, try adding try/catch calls around the life calls.

```
    if (mob_list)
        for(var/atom/m in mob_list)
            if(m)
                try
                    m:Life()
                catch(var/exception/e)
                    continue
                scheck()
                continue
            mob_list -= m
```

**Comment by AliiTea** [comment]
:+1:

**Comment by TDcoolguy300** [comment]
¯_(ツ)_/¯

**Pull request merged**